lib/shoryuken/manager.rb

```diff
@@ -19,8 +19,11 @@ def initialize(fetcher, polling_strategy)
       @polling_strategy = polling_strategy
 
       @heartbeat = Concurrent::TimerTask.new(run_now: true,
-                                             execution_interval: HEARTBEAT_INTERVAL,
-                                             timeout_interval: 60) { dispatch }
+                                             execution_interval: HEARTBEAT_INTERVAL) { @pool.post { dispatch } if @dispatching.false? }
```

Comment:
That will probably do it. It reuses the same worker pool for dispatching (which I don't really think is problematic) and the `@pool.post { ... }` will return instantly, meaning the timeout never occurs (because of the implementation, you can't actually turn off the timeout code in `TimerTask`). 

And this still lets you use `TimerTask` as a supervisor-like solution for dispatch, ensuring it stays executing, even if there is an error.